lib/AST/CASTBridging.cpp

```diff
@@ -303,311 +317,321 @@ void *BraceStmt_create(void *ctx, void *lbloc, BridgedArrayRef elements,
+  ASTContext &context = convertASTContext(cContext);
+  SourceLoc tyLoc = convertSourceLoc(cTypeLoc);
+  return new (context) MetatypeTypeRepr((TypeRepr *)baseType, tyLoc);
 }
 
-void *ProtocolTypeRepr_create(void *ctx, void *baseType, void *protoLoc) {
-  ASTContext &Context = *static_cast<ASTContext *>(ctx);
-  SourceLoc protocolLoc = getSourceLocFromPointer(protoLoc);
-  return new (Context) ProtocolTypeRepr((TypeRepr *)baseType, protocolLoc);
+void *ProtocolTypeRepr_create(BridgedASTContext cContext, void *baseType, BridgedSourceLoc cProtoLoc) {
+  ASTContext &context = convertASTContext(cContext);
+  SourceLoc protoLoc = convertSourceLoc(cProtoLoc);
+  return new (context) ProtocolTypeRepr((TypeRepr *)baseType, protoLoc);
 }
 
-void *PackExpansionTypeRepr_create(void *ctx, void *base, void *repeatLoc) {
-  ASTContext &Context = *static_cast<ASTContext *>(ctx);
-  return new (Context) PackExpansionTypeRepr(
-      getSourceLocFromPointer(repeatLoc), (TypeRepr *)base);
+void *PackExpansionTypeRepr_create(BridgedASTContext cContext, void *base, BridgedSourceLoc cRepeatLoc) {
+  ASTContext &context = convertASTContext(cContext);
+  return new (context) PackExpansionTypeRepr(
+      convertSourceLoc(cRepeatLoc), (TypeRepr *)base);
 }
 
-void *TupleTypeRepr_create(void *ctx, BridgedArrayRef elements, void *lParenLoc,
-                           void *rParenLoc) {
-  ASTContext &Context = *static_cast<ASTContext *>(ctx);
-  SourceLoc lParen = getSourceLocFromPointer(lParenLoc);
-  SourceLoc rParen = getSourceLocFromPointer(rParenLoc);
+void *TupleTypeRepr_create(BridgedASTContext cContext, BridgedArrayRef elements, BridgedSourceLoc cLParenLoc,
+                           BridgedSourceLoc cRParenLoc) {
+  ASTContext &context = convertASTContext(cContext);
+  SourceLoc lParen = convertSourceLoc(cLParenLoc);
+  SourceLoc rParen = convertSourceLoc(cRParenLoc);
   SmallVector<TupleTypeReprElement, 8> tupleElements;
-  for (auto element : getArrayRef<BridgedTupleTypeElement>(elements)) {
+  for (auto element : convertArrayRef<BridgedTupleTypeElement>(elements)) {
     TupleTypeReprElement elementRepr;
-    elementRepr.Name = Identifier::getFromOpaquePointer(element.Name);
-    elementRepr.NameLoc = getSourceLocFromPointer(element.NameLoc);
-    elementRepr.SecondName =
-        Identifier::getFromOpaquePointer(element.SecondName);
-    elementRepr.SecondNameLoc = getSourceLocFromPointer(element.SecondNameLoc);
-    elementRepr.UnderscoreLoc = getSourceLocFromPointer(element.UnderscoreLoc);
-    elementRepr.ColonLoc = getSourceLocFromPointer(element.ColonLoc);
+    elementRepr.Name = convertIdentifier(element.Name);
+    elementRepr.NameLoc = convertSourceLoc(element.NameLoc);
+    elementRepr.SecondName = convertIdentifier(element.SecondName);
+    elementRepr.SecondNameLoc = convertSourceLoc(element.SecondNameLoc);
+    elementRepr.UnderscoreLoc = convertSourceLoc(element.UnderscoreLoc);
+    elementRepr.ColonLoc = convertSourceLoc(element.ColonLoc);
```

Comment:
Not for you but: Wow, is there no initializer for this? 😳

Reply:
`TupleTypeReprElement` is a simple struct with just a `TypeRepr` constructor. TBH I'm not sure adding one would really help anyway 🤷